modules/account-lib/src/coin/sol/walletInitializationBuilder.ts

```diff
@@ -1,37 +1,65 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { BuildTransactionError } from '../baseCoin/errors';
 import { Transaction } from './transaction';
 import { TransactionBuilder } from './transactionBuilder';
-import { Transaction as SolTransaction } from '@solana/web3.js';
+import { isValidAmount, isValidPublicKey } from './utils';
 import { TransactionType } from '../baseCoin';
+import { WalletInit } from './iface';
+import { InstructionBuilderTypes } from './constants';
 
 export class WalletInitializationBuilder extends TransactionBuilder {
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
-
   protected get transactionType(): TransactionType {
     return TransactionType.WalletInitialization;
   }
 
-  /** @inheritdoc */
-  protected buildSolTransaction(): SolTransaction {
-    throw new NotImplementedError('buildImplementation not implemented');
-  }
+  /**
+   *  Creates a nonce account for an address
+   *
+   * @param {string} fromAddress - the address whos gonna own the nonce
+   * @param {string} nonceAddress - the nonce address to be created
+   * @param {string} authAddress - the address whos gonna own the nonce
+   * @param {string} amount - the amount sent to cover the rent of the nonce address
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  walletInit(fromAddress: string, nonceAddress: string, authAddress: string, amount: string): this {
```

Comment:
traditional builder pattern is to split this up into multiple functions, 1 for each input.
`build()` then returns the resulting, immutable transaction

Reply:
initial impl doesn't need to have a different `authAddress` from `fromAddress`. remove for now, we will add back if needed